File: hdbank-research-model_service-c9279bfc573f/api/utils/response/BaseResponse.py
from fastapi.responses import JSONResponse


class BaseResponse:
    def __init__(self, content, status=200):
        self.content = content
        self.status = status

    def success(self):
        return JSONResponse(
            content={
                "data": self.content,
                "error": None,
            }
        )

    def error(self):
        return JSONResponse(
            content={
                "error": self.content,
                "data": None,
            }
        )

File: hdbank-research-api_service-9e3c73aacb59/src/utils/generics/base.response.ts
export class BaseResponse<T> {
    data: Nullable<T | null>;
    error: any;

    private constructor(data: Nullable<T>, error: any) {
        this.data = data;
        this.error = error;
    }

    static of<T>(data: T): BaseResponse<T> {
        return new BaseResponse(data, null);
    }

    static error<T>(error: any): BaseResponse<Nullable<T>> {
        return new BaseResponse(null, error);
    }

    static ok<T>(): BaseResponse<Nullable<T>> {
        return new BaseResponse(null, null);
    }
}

type Nullable<T> = T | null;
